hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java

```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * Create a new {@link WritableAccountStore} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableAccountStore(@NonNull final WritableStates states) {
+        super(states);
+        requireNonNull(states);
+
+        this.accountState = states.get(TokenServiceImpl.ACCOUNTS_KEY);
+        this.aliases = states.get(TokenServiceImpl.ALIASES_KEY);
+    }
+
+    /**
+     * Persists a new {@link Account} into the state, as well as exporting its ID to the transaction
+     * receipt.
+     *
+     * @param account - the account to be added to modifications in state.
+     */
+    public void put(@NonNull final Account account) {
+        Objects.requireNonNull(account);
+        accountState.put(EntityNumVirtualKey.fromLong(account.accountNumber()), Objects.requireNonNull(account));
+    }
+
+    /**
+     * Persists a new alias linked to the account persisted to state
+     *
+     * @param alias - the alias to be added to modifications in state.
+     * @param accountNum - the account number to be added to modifications in state.
+     */
+    public void putAlias(@NonNull final String alias, @NonNull final long accountNum) {
+        Objects.requireNonNull(alias);
+        aliases.put(alias, new EntityNumValue(accountNum));
+    }
+
+    /** Commits the changes to the underlying data storage. */
+    public void commit() {
+        ((WritableKVStateBase) accountState).commit();
+        ((WritableKVStateBase) aliases).commit();
+    }
+
+    /**
+     * Returns the {@link Account} with the given number. If no such account exists, returns {@code
+     * Optional.empty()}
+     *
+     * @param accountID - the id of the Account to be retrieved.
+     */
+    @NonNull
+    public Optional<Account> get(final AccountID accountID) {
+        requireNonNull(accountID);
+        final var account = getAccountLeaf(accountID);
+        return Optional.ofNullable(account);
+    }
+
+    /**
+     * Returns the {@link Account} with the given number using {@link
+     * WritableKVState#getForModify(Object K)}}. If no such account exists, returns {@code
+     * Optional.empty()}
+     *
+     * @param id - the number of the account to be retrieved.
+     */
+    @NonNull
+    public Optional<Account> getForModify(final AccountID id) {
+        // Get the account number based on the account identifier. It may be null.
+        final var accountOneOf = id.account();
+        final Long accountNum =
+                switch (accountOneOf.kind()) {
+                    case ACCOUNT_NUM -> accountOneOf.as();
+                    case ALIAS -> {
+                        final Bytes alias = accountOneOf.as();
+                        if (alias.length() == EVM_ADDRESS_LEN && isMirror(alias)) {
+                            yield fromMirror(alias);
+                        } else {
+                            final var entityNum = aliases.getForModify(alias.asUtf8String());
```

Comment:
Doesn't `aliases.get()` make more sense here than `aliases.getForModify()`?

Reply:
Not sure. Just thought may be that helps with performance, since FcHashMap also has `getForModify`. Do you think it should not matter ?

Reply:
I think in general modifying an account won't imply changing an alias, and since `getForModify()` is more expensive than `get()`, I'd lean toward the latter.

Reply:
I see .. That makes sense.

Reply:
Changed